src/main/java/org/broadinstitute/consent/http/cloudstore/GCSService.java

```diff
@@ -138,6 +143,23 @@ public Map<BlobId, InputStream> getDocuments(List<BlobId> blobIds) throws NotFou
     }
   }
 
+  /**
+   * Read JSON file from GCS bucket
+   *
+   * @param blobIdName String value of the document blob id name
+   * @param valueType  Class type of the object to be read
+   * @return Object of the specified type
+   */
+  public <T> T readJsonFileFromBucket(String blobIdName, Class<T> valueType) throws IOException {
+    try (InputStream is = getDocument(blobIdName)) {
+      InputStreamReader reader = new InputStreamReader(is);
+      return GSON.fromJson(reader, valueType);
```

Comment:
This is a little convoluted in that the actual document source is `byte[]` but `getDocument()` converts it to an `InputStream` which this code converts back to a GSON value. It would be more efficient to have a `getDocument()` API that returns `byte[]`
```suggestion
    try {
      byte[] bytes = getDocumentBytes(blobIdName);
      return GSON.fromJson(new String(bytes), valueType);
```